blockstack_client/rpc.py

```diff
@@ -920,7 +919,31 @@ def GET_name_info( self, ses, path_info, name ):
             if 'not found' in name_rec['error'].lower():
                 return self._reply_json({'status': 'available'}, status_code=404)
             elif 'failed to load subdomain' in name_rec['error'].lower():
-                return self._reply_json({'status': 'available'}, status_code=404)
+                # handle redirection to specified resolver
+                _, _, domain_name = blockstackd_scripts.is_address_subdomain(name)
+                domain_rec = blockstackd_client.get_name_record(domain_name, include_history = False,
+                                                                hostport = blockstackd_url)
+                if 'error' in domain_rec or 'zonefile' not in domain_rec:
+                    return self._reply_json(
+                        {'status': 'available', 'more': 'failed to lookup parent domain'}, status_code=404)
+                domain_zf_txt = base64.b64decode(domain_rec['zonefile'])
+                domain_zf_json = zonefile.decode_name_zonefile(domain_name, domain_zf_txt, allow_legacy=False)
+                matching_uris = [ x['target'] for x in domain_zf_json['uri'] if x['name'] == '_resolver' ]
+                if len(matching_uris) == 0:
+                    return self._reply_json(
+                        {'status': 'available',
+                         'more': 'failed to find parent domain\'s resolver'}, status_code=404)
```

Comment:
What happens if there is more than one `_resolver`?  Does this warrant another type of error message?

Also, if we're going to include ancillary messages, should we classify them somehow?  I.e. `info`, `warning`, `error` instead of just `more`?

Reply:
If there's more than one `_resolver`, this will pick an unspecified one of them. We could order them by zonefile priority and select that way, but I believe that'd be over-engineering.

In terms of including ancillary messages, I think they're helpful if someone is trying to debug their subdomain registrar setup. I'm happy to call it `message` instead (that's used in other places), but I'm wary of trying to classify it, especially in that code -- the presence of the `error` key in a response object is used to indicate failure in a lot of places in our codebase.